### PR TITLE
enter: escape --workdir argument #181

### PR DESCRIPTION
**Pull request by nm004**

### CODE REVIEW

**Comment by nm004** [comment]
There could be a better approach than escapaing all chars, I think.

File: distrobox-enter [hunk 1]
```diff
@@ -217,8 +217,9 @@ generate_command() {
 	# Entering container using our user and workdir.
 	# Start container from working directory. Else default to home. Else do /.
 	# pass distrobox-enter path, it will be used in the distrobox-export tool.
+	workdir=$(printf "%s" "${PWD:-${HOME:-/}}" | sed 's/./\\&/g')
 	result_command="${result_command}
-		--workdir=${PWD:-${HOME:-"/"}}
+		--workdir=${workdir}
```
Comment (89luca89) [inline distrobox-enter:222]: Seems to be enough to just quote (escaping) the workdir and it works with spaces and special chars:

`--workdir=\"${PWD:-${HOME:-"/"}}\"`

**Review by 89luca89 — CHANGES_REQUESTED** [review]

**Comment by 89luca89** [comment]
> There could be a better approach than escapaing all chars, I think.

Hi @nm004 thanks a lot for the contribution
I've commented another way to make the workdir play nicely with special chars, from a quick test it works with the examples you reported on #182 
can you check also? thanks! :+1:

File: distrobox-enter (hunk 1 above)
Comment (89luca89) [reply distrobox-enter:222]: I'll correct myself, this won't cover 1 use-case, the presence of quotes in the path name. using this should solve our prolems:

```sh
	workdir="$(echo "${PWD:-${HOME:-"/"}}" | sed -e 's/"/\\\"/g' )"
	result_command="${result_command}
		--workdir=\"${workdir}\"
```

**Review by 89luca89 — CHANGES_REQUESTED** [review]

File: distrobox-enter (hunk 1 above)
Comment (nm004) [reply distrobox-enter:223]: @89luca89 Thank you for your reply.

It works perfectly.

**Comment by 89luca89** [comment]
@nm004 thanks, you just need to run a `shfmt -s` on the script before pushing again so that it solves the formatting issue that are blocking the shellcheck action :+1:

**Comment by nm004** [comment]
@89luca89

> @nm004 thanks, you just need to run a `shfmt -s` on the script before pushing again so that it solves the formatting issue that are blocking the shellcheck action +1

oh, ok thx. Is it ok now?

**Review by 89luca89 — APPROVED** [review]

**Comment by 89luca89** [comment]
Yep thanks for the contribution! Merging!

**Comment by nm004** [comment]
Thanks :)

**Pull request merged**